### PR TITLE
Change HA `hold_mode` to `preset_mode`

### PR DESCRIPTION
**Pull request by rikroe**
This PR changes `hold_mode*` to `preset_mode*` for climate devices like the [Lidl Silvercrest Thermostat](https://www.zigbee2mqtt.io/devices/368308_2010.html).

Since 2022.2 (https://github.com/home-assistant/core/pull/66062) `hold_mode` (and `away_mode` which I didn't touch yet) is deprecated and will be removed in 2022.9.

Regarding `away_mode` more coding might be necessary - for HA it will be just another `preset` send, i.e. `/set/preset` `away`.
However my JS knowledge is very limited and I don't feel comfortable editing more than a few strings.

EDIT: Ok, changing `/set/fan_mode` to `/set/preset` apparently broke the tests. Maybe this needs more thought than expected. 
Any help would be appreciated.

I also realized that the `preset_modes` supplied by my thermostat are not the "default" ones in HA, i.e. those with translations and icons...

### CODE REVIEW

**Comment by Koenkk** [comment]
> However my JS knowledge is very limited and I don't feel comfortable editing more than a few strings.

Can you try making the change? I can review it.

File: lib/extension/homeassistant.ts [hunk 1]
```diff
@@ -1251,7 +1243,7 @@ export default class HomeAssistant extends Extension {
             }
 
             if (payload.preset_mode_command_topic) {
```
Comment (rikroe) [inline lib/extension/homeassistant.ts:1245]: Alright, I at least found what caused my confusion :)

There is a preset mode in both [MQTT Fan](https://www.home-assistant.io/integrations/fan.mqtt/#preset_mode_command_template) as well as [MQTT HVAC](https://www.home-assistant.io/integrations/climate.mqtt/#preset_mode_command_template) both support `preset_mode*`, however with slightly different settings.

I think we would need to separate setting the command topic here based on the type of device (fan or climate), something like `discoveryEntry.type == "climate"` which seems to be unavailable here.

Comment (rikroe) [reply lib/extension/homeassistant.ts:1245]: Thanks, will have a look at it soon!

**Comment by Koenkk** [comment]
Did some updates, should be good now. Will look into the `away_mode`.

**Comment by Koenkk** [comment]
Partially addresses https://github.com/Koenkk/zigbee2mqtt/issues/11738

**Pull request merged**